File: Arrays_And_Hashing/PythonSolutions/FindTheDifference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        

        seen = {}
        seen2= {}
        for char in s:
            seen[char] = seen.get(char,0)+1

        for char in t:
            seen2[char] = seen2.get(char,0)+1

        for char in seen2:
            if seen2[char] > seen.get(char,0):
                return char 
     
        
        return 

File: Arrays_And_Hashing/JavaSolutions/FindtheDifference.java
package Arrays_And_Hashing.JavaSolutions;
import java.util.HashMap;

class Solution {
    public char findTheDifference(String s, String t) {
        HashMap<Character, Integer> seen = new HashMap<>();
        HashMap<Character, Integer> seen2 = new HashMap<>();

        for (char c : s.toCharArray()) {
            seen.put(c, seen.getOrDefault(c, 0) + 1);
        }

        for (char c : t.toCharArray()) {
            seen2.put(c, seen2.getOrDefault(c, 0) + 1);
        }

        for (char c : seen2.keySet()) {
            if (seen2.get(c) > seen.getOrDefault(c, 0)) {
                return c;
            }
        }

        return '\0'; // default return, should never hit due to constraints
    }
}
